src/web/mjs/connectors/CrunchyManga.mjs

```diff
@@ -88,9 +90,36 @@ export default class CrunchyManga extends Crunchyroll {
     _decryptImage(encrypted, key) {
         // create a view for the buffer
         let decrypted = new Uint8Array(encrypted);
-        for(let index in decrypted) {
+        for (let index in decrypted) {
             decrypted[index] ^= key;
         }
         return decrypted;
     }
+
+    async _getMangaFromURI(uri) {
+        await this._login();
+        const request = new Request(uri, this.requestOptions);
+        let response = await this.fetchDOM(request);
+        let responseData = await fetch(request);
+        let responseDataText = await responseData.text();
+
+        let title;
+        let seriesId;
+        let seriesIdList = [...responseDataText.matchAll(/series_id:(\d+)/g)];
```

Comment:
```js
/series_id:(\d+)/g.exec(responseDataText)
```
seems more reasonable since you only ever use the first match anyways
(the variable name would then be changed since its no longer a list)

(the if would have to changed to `if (seriesIdList) {`)

Reply:
Couldn't figure this out so left it as is

Reply:
```js
        let hasSeriesId = /series_id:(\d+)/g.exec(responseDataText)

        if (hasSeriesId) {
            title = response.querySelector('div.queue-wrapper > a').title.trim()
            seriesId = Number(hasSeriesId[1])
        }
```